crates/gateway-types/src/error.rs

```diff
@@ -33,10 +33,24 @@ impl std::fmt::Display for StarknetError {
     }
 }
 
-/// Represents starknet specific error codes reported by the sequencer.
+/// Represents a starknet error code reported by the sequencer.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum StarknetErrorCode {
+    Known(KnownStarknetErrorCode),
+    Unknown(String),
+}
```

Comment:
Did you consider flattening these? As in something like:

```rust
pub enum GatewayErrorCode {
  #[serde(rename = "StarknetErrorCode.BLOCK_NOT_FOUND")]
  BlockNotFound,
  ... etc
  Other(String),
}
```

Not sure if that works, but it would be less types? And then removing error types we don't care about at all and just letting them fall through.. although no harm in keeping the known ones I suppose in case we do want them.

Reply:
Nah, `serde(other)` works only for unit variants...

Reply:
Sadness. Seems like such a strange thing to me.. I wonder if serde is actually the correct tool for our gateway types. Anyway, problem for another day.